migrations/20220905173159__rubrics__create.sql

```diff
@@ -0,0 +1,58 @@
+CREATE TABLE IF NOT EXISTS rubrics (
+    id BIGSERIAL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    starting_points DOUBLE PRECISION NOT NULL,
+    max_points DOUBLE PRECISION NOT NULL,
+    min_points DOUBLE PRECISION NOT NULL,
+    modified_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    deleted_at TIMESTAMP WITH TIME ZONE
+);
+
+CREATE TABLE IF NOT EXISTS rubric_items (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_id BIGINT NOT NULL REFERENCES rubrics(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    number BIGINT NOT NULL,
+    points DOUBLE PRECISION NOT NULL,
+    short_text TEXT,
+    description TEXT,
+    staff_instructions TEXT,
+    key_binding TEXT,
+    deleted_at TIMESTAMP WITH TIME ZONE
+);
+
+CREATE INDEX IF NOT EXISTS rubric_items_rubric_id ON rubric_items(rubric_id);
+
+CREATE TABLE IF NOT EXISTS rubric_gradings (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_id BIGINT NOT NULL REFERENCES rubrics(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    computed_points DOUBLE PRECISION NOT NULL,
+    adjust_points DOUBLE PRECISION NOT NULL,
+    starting_points DOUBLE PRECISION NOT NULL,
+    max_points DOUBLE PRECISION NOT NULL,
+    min_points DOUBLE PRECISION NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS rubric_grading_items (
+    id BIGSERIAL PRIMARY KEY,
+    rubric_grading_id BIGINT NOT NULL REFERENCES rubric_gradings(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    rubric_item_id BIGINT NOT NULL REFERENCES rubric_items(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    score DOUBLE PRECISION NOT NULL DEFAULT 1,
+    points DOUBLE PRECISION NOT NULL,
+    short_text TEXT,
+    note TEXT
+);
+
+-- TODO Consider other indices
+
+ALTER TABLE rubric_grading_items DROP CONSTRAINT IF EXISTS rubric_grading_items_rubric_grading_id_rubric_item_id_key;
```

Comment:
It's to make it idempotent. It should only exist if you run the migration twice, and in that case this statement avoids an error. It's like add if not exists, but constraints don't have that option.

Reply:
Gotcha. @mwest1066 this is something I haven't really thought about in a while, but why do we care if migrations are idempotent? When would they ever be run twice (except for situations where we need to manually rerun portions, but even then, we shouldn't have to design migrations as a whole to be idempotent).

We *shouldn't* ever be running migrations twice, but if we do, giving ourselves a window (however short) where constraints aren't enforced seems deeply undesirable.

Reply:
I believe (and I may be wrong) that, since these two are running in the same transaction the table would be locked until the constraint is added back. That said, I don't mind removing it.

Reply:
Removed in ad969bcb5.